BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -0,0 +1,1041 @@
+                    {
+                        currentStatus = "Maneuvering (Kill Angular Velocity)";
+
+                        while (UnderTimeLimit() && targetVessel != null && !complete)
+                        {
+                            complete = AngularVelocity(vessel, targetVessel) < firingAngularVelocityLimit / 2;
+                            fc.attitude = Vector3.ProjectOnPlane(RelVel(vessel, targetVessel), FromTo(vessel, targetVessel)).normalized * -1;
+                            fc.throttle = !complete ? 1 : 0;
+
+                            yield return new WaitForFixedUpdate();
+                        }
+                    }
+                    else
+                    {
+                        if (hasPropulsion)
+                        {
+                            if (currentRange < minRange)
+                            {
+                                currentStatus = "Maneuvering (Drift Away)";
+
+                                Vector3 toTarget;
+                                fc.throttle = 0;
+
+                                while (UnderTimeLimit() && targetVessel != null && !complete)
+                                {
+                                    toTarget = FromTo(vessel, targetVessel);
+                                    complete = toTarget.magnitude > minRange;
+                                    fc.attitude = toTarget.normalized;
+
+                                    yield return new WaitForFixedUpdate();
+                                }
+                            }
+                            else
+                            {
+                                currentStatus = "Maneuvering (Drift)";
+                                fc.throttle = 0;
+                                fc.attitude = Vector3.zero;
+                            }
+                        }
+                        else
+                        {
+                            currentStatus = "Stranded";
+                            fc.throttle = 0;
+                            fc.attitude = Vector3.zero;
+                        }
+
+                        yield return new WaitForSeconds(updateInterval);
+                    }
+                }
+            }
+            else
+            {
+                // Idle
+
+                if (hasWeapons)
+                    currentStatus = "Idle";
+                else
+                    currentStatus = "Idle (Unarmed)";
+
+                fc.throttle = 0;
+                fc.attitude = Vector3.zero;
+
+                yield return new WaitForSeconds(updateInterval);
+            }
+        }
+
+        #endregion Actual AI Pilot
+
+        #region Utility Functions
+
+        private bool CheckWithdraw()
+        {
+            var nearest = BDATargetManager.GetClosestTarget(weaponManager);
+            if (nearest == null) return false;
+
+            return Mathf.Abs(RelVel(vessel, nearest.Vessel).magnitude) < 200;
```

Comment:
`magnitude` is already positive, so `Abs` is unnecessary, however, this should use `sqrMagnitude` instead.